java/code/src/com/redhat/rhn/manager/setup/ProductSyncManager.java

```diff
@@ -495,10 +492,11 @@ private SetupWizardProductDto convertProduct(final MgrSyncProductDto productIn)
      * Find a product for any given ident by looking through base and their addons.
      *
      * @param ident ident of a product
+     * @param productsWithAddons the products list with addons
      * @return the {@link SetupWizardProductDto}
      */
-    private SetupWizardProductDto findProductByIdent(String ident) {
-        for (SetupWizardProductDto p : getBaseProducts()) {
+    private SetupWizardProductDto findProductByIdent(String ident, List<SetupWizardProductDto> productsWithAddons) {
+        for (SetupWizardProductDto p : productsWithAddons) {
             if (p.getIdent().equals(ident)) {
                 return p;
             }
```

Comment:
GitHub is not letting me to put the comment in the correct line :man_facepalming: Anyway, below here on line 503, I think you can remove the loop over `p.getAddonProducts()` since now the addons are part of the main list.

Reply:
That's a good news, thanks! I'm wondering if maybe I can remove at all `addonProducts` https://github.com/uyuni-project/uyuni/blob/3b7d8d34dff8809a97874f35421df952d1be8150/java/code/src/com/redhat/rhn/frontend/dto/SetupWizardProductDto.java#L249  ..what do you think?

Reply:
same here ? https://github.com/uyuni-project/uyuni/blob/3b7d8d34dff8809a97874f35421df952d1be8150/java/code/src/com/redhat/rhn/domain/product/SUSEProductSet.java#L39

Reply:
From the DTO maybe we can since, it seems to be used only in the method you changed so I think it's safe to "flatten" the representation. In the `SUSEProductSet` business object I think it's needed.

Reply:
After 1:1 conversation, we decided it's not worth to remove the `addonProducts` field from the DTO.